irohad/torii/processor/impl/transaction_processor_impl.cpp

```diff
@@ -32,22 +33,81 @@ namespace iroha {
         std::shared_ptr<StatelessValidator> validator)
         : pcs_(std::move(pcs)), validator_(std::move(validator)) {
       log_ = logger::log("TxProcessor");
+
+      // insert all txs from proposal to proposal set
+      pcs_->on_proposal().subscribe([this](model::Proposal proposal) {
+        for (auto tx : proposal.transactions) {
+          proposal_set_.insert(tx.tx_hash.to_string());
+          TransactionResponse response;
+          response.tx_hash = tx.tx_hash.to_string();
+          response.current_status =
+              TransactionResponse::STATELESS_VALIDATION_SUCCESS;
+          notifier_.get_subscriber().on_next(
+              std::make_shared<model::TransactionResponse>(response));
+        }
+      });
+
+      // move commited txs from proposal to candidate map
+      pcs_->on_commit().subscribe([this](
+          rxcpp::observable<model::Block> blocks) {
+        blocks.subscribe(
+            // on next..
+            [this](model::Block block) {
+              for (auto tx : block.transactions) {
+                if (this->proposal_set_.count(tx.tx_hash.to_string())) {
+                  proposal_set_.erase(tx.tx_hash.to_string());
+                  candidate_set_.insert(tx.tx_hash.to_string());
+                  TransactionResponse response;
+                  response.tx_hash = tx.tx_hash.to_string();
+                  response.current_status =
+                      model::TransactionResponse::STATEFUL_VALIDATION_SUCCESS;
```

Comment:
Maybe when tx processor receives the transaction Response is already COMMITED ?

Reply:
In my understanding while the chain is preparing by supplying blocks to subject, transactions are not committed. But when the chain is ready and on complete is invoked, we should change status from stateful valid to committed